tests/unit/customizations/eks/test_update_kubeconfig.py

```diff
@@ -217,10 +217,16 @@ def setUp(self):
         self._session.create_client.return_value = self._mock_client
         self._session.profile = None
 
-        self._client = EKSClient(self._session, "ExampleCluster", None)
+        self._client = EKSClient(self._session, cluster_name="ExampleCluster", role_arn=None)
+
+    def test_init_client_with_non_supported_parsed_args(self):
+        client = EKSClient(self._session, cluster_name="ClusterName", role_arn=None, non_supported_attr="random value")
+        self.assertIn('_cluster_name', vars(client))
```

Comment:
If we move to just accepting the `parsed_args` and not expanded out as keyword arguments, we should be able to remove this test. In general, we should avoid asserting against private members (e.g. members that begin with underscore) as there is no guarantee that interface is stable.